### ISSUE TITLE
Error when trying to get empty list of open orders

### ISSUE THREAD

**Issue by IldarDavletyarov** [stoqey/ib#124]
If the list of open orders is **empty**, an error is thrown

code: `await ibApiNext.getAllOpenOrders();`

```
/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/util/createErrorClass.ts:13
    instance.stack = new Error().stack;
                     ^
Error
    at _super (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/util/createErrorClass.ts:13:22)
    at new EmptyErrorImpl (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/util/EmptyError.ts:26:3)
    at Object.complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/lastValueFrom.ts:72:18)
    at Object.complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:194:14)
    at SafeSubscriber.Subscriber._complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at SafeSubscriber.Subscriber.complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:106:12)
    at OperatorSubscriber.Subscriber._complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:132:24)
    at OperatorSubscriber.Subscriber.complete (/Users/ildar/DIPLOM/ibkr-signal/node_modules/rxjs/src/internal/Subscriber.ts:106:12) {
  message: 'no elements in sequence'
}
```

**Comment by mfrener** [comment]
Can you post some code how to reproduce this?
Tested "node ./dist/tools/open-orders.js" - when no order are open it doesn't do anything (waiting for an order), otherwise it returns the list. Also modified open-order.ts to do "this.printObject(await this.api.getAllOpenOrders());", making the sure 'await' doesn't cause any side-effect, but works as well.

**Comment by mfrener** [comment]
Just got it reproduced: it can happen the "end" event without any prior order event, although TWS and Gateway seem to behave differently. Anyhow.. we miss the default return value on our Observable -> Promise conversion, so when Observable completes but has not emitted any values, the Promise throws and error because it doesn't have any value to resolve.
Fix incomming.

**Comment by mfrener** [comment]
PR open https://github.com/stoqey/ib/pull/126

**Comment by IldarDavletyarov** [comment]
Thank you very much for the quick response!

**Comment by mfrener** [comment]
Fixed with v1.2.14
